### PR TITLE
Moved SearchableListMixin to wagtailadmin

### PR DESCRIPTION
**Pull request by kaedroho**
And add wagtailsearch support.

As suggested by @ababic in https://github.com/wagtail/wagtail/pull/3774#discussion_r141448118

### CODE REVIEW

File: wagtail/wagtailadmin/views/mixins.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+                queryset = search_backend.search(q, queryset, fields=self.search_fields)
+            else:
+                filters = {
+                    field + '__icontains': q
+                    for field in self.search_fields or []
+                }
+
+                queryset = queryset.filter(**filters)
+
+        return queryset
```
Comment (BertrandBordage) [inline wagtail/wagtailadmin/views/mixins.py]: Defining `queryset` is useless here, we can use `return` directly and avoid the `else:`.

File: wagtail/wagtailadmin/views/mixins.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+class SearchableListMixin(object):
+    search_box_placeholder = _("Search")
+    search_fields = None
+
+    def get_search_form(self):
+        if self.request.GET.get('q'):
+            return SearchForm(self.request.GET, placeholder=self.search_box_placeholder)
+        else:
+            return SearchForm(placeholder=self.search_box_placeholder)
```
Comment (BertrandBordage) [inline wagtail/wagtailadmin/views/mixins.py]: To reduce redundancy, can be replaced with:
```python
def get_search_form(self):
    return SearchForm(self.request.GET if self.request.GET.get('q') else None,
                      placeholder=self.search_box_placeholder)
```

File: wagtail/wagtailadmin/views/mixins.py (hunk 1 above)
Comment (kaedroho) [reply wagtail/wagtailadmin/views/mixins.py]: I'm not sure what you mean, this method should always return a queryset.

**Comment by kaedroho** [comment]
Thanks for the review @BertrandBordage!

File: wagtail/wagtailadmin/views/mixins.py (hunk 1 above)
Comment (BertrandBordage) [reply wagtail/wagtailadmin/views/mixins.py]: Yes, it returns a queryset, but it does not have to define a `queryset` variable, querysets can be returned directly.

**Pull request merged**